### PR TITLE
Auto-commit disabled. Concurrent message consumption. Graceful shutdown.

### PR DESCRIPTION
**Pull request by catalin-ilea**
* Offset commit is now done manually
* Added stream functionality
* Graceful shutdown: destroying consumers instance when SIGTERM received, not relying on the proxy to expire consumers which means partitions are rebalanced more quickly

### CODE REVIEW

**Comment by kovacshuni** [comment]
+1

**Comment by tamas-molnar** [comment]
lgtm

**Comment by Nesze** [comment]
+1

File: src/main/java/com/ft/message/consumer/proxy/MessageQueueProxyServiceImpl.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public URI createConsumerInstance() {
             if (queueIsNotEmpty()) {
                 builder.header("Host", configuration.getQueue());
             }
-            clientResponse = builder.post(ClientResponse.class, "{\"auto.offset.reset\": \"smallest\", \"auto.commit.enable\": \"true\"}");
+            clientResponse = builder.post(ClientResponse.class, String.format("{\"auto.offset.reset\": \"%s\", \"auto.commit.enable\": \"true\"}", configuration.getOffsetReset()));
```
Comment (tosan88) [inline src/main/java/com/ft/message/consumer/proxy/MessageQueueProxyServiceImpl.java]: Should be `"auto.commit.enable\": \"false\"`

Comment (catalin-ilea) [inline src/main/java/com/ft/message/consumer/proxy/MessageQueueProxyServiceImpl.java]: Indeed, good catch. Updated.

**Comment by tosan88** [comment]
:+1:

**Pull request merged**